src/main/cms/cms_menu_imu.c

```diff
@@ -126,16 +140,108 @@ static long cmsx_rateProfileIndexOnChange(displayPort_t *displayPort, const void
     return 0;
 }
 
-static long cmsx_PidRead(void)
+static long cmsx_PidAdvancedRead(void)
 {
 
     const pidProfile_t *pidProfile = pidProfiles(pidProfileIndex);
+
     feathered_pids = pidProfile->feathered_pids;
     i_decay = pidProfile->i_decay;
     errorBoost = pidProfile->errorBoost;
     errorBoostLimit = pidProfile->errorBoostLimit;
     errorBoostYaw = pidProfile->errorBoostYaw;
     errorBoostLimitYaw = pidProfile->errorBoostLimitYaw;
+    integralHalfLife = pidProfile->integral_half_life;
+    integralHalfLifeYaw = pidProfile->integral_half_life_yaw;
+    QuickFlashRelax = pidProfile->QuickFlashRelax;
+    QuickFlashRelaxYaw = pidProfile->QuickFlashRelaxYaw;
+    QuickFlashRelaxCutoff = pidProfile->QuickFlashRelaxCutoff;
+    QuickFlashRelaxType = pidProfile->QuickFlashRelaxType;
+    itermWindup = pidProfile->itermWindupPointPercent;
+    iDecayV2 = pidProfile->iDecayV2;
+    iDecayCutoff = pidProfile->iDecayCutoff;
+
+    return 0;
+}
+
+static long cmsx_PidAdvancedOnEnter(void)
+{
+    pidProfileIndexString[1] = '0' + tmpPidProfileIndex;
+    cmsx_PidAdvancedRead();
+
+    return 0;
+}
+
+static long cmsx_PidAdvancedWriteback(const OSD_Entry *self)
+{
+    UNUSED(self);
+
+    pidProfile_t *pidProfile = currentPidProfile;
+
+    pidProfile->feathered_pids = feathered_pids;
+    pidProfile->errorBoost = errorBoost;
+    pidProfile->errorBoostLimit = errorBoostLimit;
+    pidProfile->errorBoostYaw = errorBoostYaw;
+    pidProfile->errorBoostLimitYaw = errorBoostLimitYaw;
+    pidProfile->i_decay = i_decay;
+    pidProfile->integral_half_life = integralHalfLife;
+    pidProfile->integral_half_life_yaw = integralHalfLifeYaw;
+    pidProfile->QuickFlashRelax = QuickFlashRelax;
```

Comment:
uhm?

Reply:
naming was off on purpose to help prevent testers from seeing what exactly they are testing for less placebo.

Reply:
the naming convention of variables is not related to their purpose.

Reply:
it is only temporary naming for the purpose of "blind testing".

Reply:
so the testers read the source code in the OSD?